### PR TITLE
Add MessageController [by geobra]

### PR DESCRIPTION
**Pull request by LNJ2**
This is a split up of geobras PR: #31.

### CODE REVIEW

**Comment by JBBgameich** [comment]
Finally it's there! Is this ready meaned for merging or should a kirigami gui for the chat be done before?

File: kaidan.pro [hunk 1]
```diff
@@ -20,15 +21,21 @@ DEFINES += BOOST_SIGNALS_NO_DEPRECATION_WARNING
 
 SOURCES += src/main.cpp \
 	src/Kaidan.cpp \
-	src/RosterController.cpp \
-	src/RosterItem.cpp
+	src/RosterContoller.cpp \
```
Comment (JBBgameich) [inline kaidan.pro]: Please don't change this back again

File: kaidan.pro [hunk 2]
```diff
@@ -20,15 +21,21 @@ DEFINES += BOOST_SIGNALS_NO_DEPRECATION_WARNING
...
+	src/persistence/MessageController.cpp \
+	src/persistence/Persistence.cpp
 
 HEADERS += src/Kaidan.h \
 	src/EchoPayload.h \
 	src/EchoPayloadParserFactory.h \
 	src/EchoPayloadSerializer.h \
-	src/RosterController.h \
-	src/RosterItem.h
+	src/RosterContoller.h \
```
Comment (JBBgameich) [inline kaidan.pro]: same here

**Comment by LNJ2** [comment]
FYI: I'm documenting/cleaning the code at the moment.

**Comment by LNJ2** [comment]
Should the Kirigami UI be included in this PR or in a new one? (I'd suggest a new one)

**Comment by JBBgameich** [comment]
I think making a new one is better, the best would be if the Sailfish ui, maybe that one from geobra could be merged at nearly the same time

**Comment by LNJ2** [comment]
Whenever a function of Persistence / MessageController is called, Kaidan crashes ...

**Pull request closed without merging**